File: src/main/java/im/tox/tox4j/core/callbacks/FriendStatusMessageCallback.java
package im.tox.tox4j.core.callbacks;

import im.tox.tox4j.annotations.NotNull;

/**
 * This event is triggered when a friend changes their status message.
 */
public interface FriendStatusMessageCallback {
  FriendStatusMessageCallback IGNORE = new FriendStatusMessageCallback() {
    @Override
    public void friendStatusMessage(int friendNumber, @NotNull byte[] message) {
    }
  };

  /**
   * @param friendNumber The friend number of the friend whose status message changed.
   * @param message The new status message.
   */
  void friendStatusMessage(int friendNumber, @NotNull byte[] message);
}

File: projects/tox4j/src/main/java/im/tox/tox4j/core/callbacks/FriendStatusMessageCallback.scala
package im.tox.tox4j.core.callbacks

import org.jetbrains.annotations.NotNull

/**
 * This event is triggered when a friend changes their status message.
 */
trait FriendStatusMessageCallback[ToxCoreState] {
  /**
   * @param friendNumber The friend number of the friend whose status message changed.
   * @param message The new status message.
   */
  def friendStatusMessage(
    friendNumber: Int, @NotNull message: Array[Byte]
  )(state: ToxCoreState): ToxCoreState = state
}
